### ISSUE TITLE
Sometimes Train_loss to become NaN on RTX 8000.

### ISSUE THREAD

**Issue by shifsa** [openvla/openvla#133]
I am fine-tuning OpenVLA using an RTX 8000 GPU. 
However, sometimes the training loss becomes NaN during the training process, even with the same parameters. 
Sometimes, the training completes without any NaN occurrences.
Since the RTX 8000 does not support bfloat16, I am using a partial implementation of float16 as mentioned in issue #110. 
Could the NaN issue still be related to not using bfloat16, as described in issue #30?

**Comment by kpertsch** [comment]
Yeah, bfloat16 is generally preferable for training workloads since it is numerically more stable. 
The easiest would be to switch to a GPU that supports bfloat16. If this is not an option you could figure out which parameters turn to NaNs and explicitly upcast these params to float32 to avoid the numerical instabilities.
